# LEARNING PYTHON
### Week 06 
#### Emma Wasserman

## ARITHMETIC

In [1]:
# Add 3 and 4
3 + 4

7

In [2]:
# Add the following numbers 3,4,5,2,18,22
3 + 4 + 5 + 2 + 18 + 22

54

In [6]:
# Subtract 100 from 1000
1000 - 100


900

In [8]:
# Multiple 3 by 100
3 * 100

300

In [9]:
#Divide 100 by 3
100 / 3

33.333333333333336

In [10]:
type(100 / 3)

float

In [11]:
# calculate 100 divided by 3 but do integer division
100 // 3

33

In [12]:
# calculate 100/3 but return the remainder 
100 % 3

1

In [14]:
# Calculate 3 to the power of 4 
3 ** 4

81

$$
\frac{(6+4)\times2}{10}\\
$$

In [3]:
# ((6+4)*2)/10
((6 + 4) * 2) / 10


2.0

## Comparisons

In [19]:
True

True

In [20]:
type(True)

bool

In [21]:
True + True

2

In [22]:
False

False

In [23]:
# Determine if 3 is larger than 5
3 > 5

False

In [24]:
# Determine if 3 is less than or equal to 3
3 <= 3

True

In [25]:
# Determine if 3 is equal to 3
3 == 3

True

In [26]:
# Determine if 3 is not equal to 3
3 != 3


False

# Python data types

In [27]:
type (3)

int

In [28]:
type (3.0)

float

In [29]:
3 == 3.0

True

In [30]:
type (1+1j)

complex

In [31]:
type('Python')

str

In [32]:
type("Python")

str

In [33]:
print("This is 'Python'")

This is 'Python'


In [34]:
type('8')

str

In [35]:
# Convert the string '8' to an integer
int('8')

8

In [36]:
float ('8')

8.0

In [37]:
float ('8') + 3

11.0

## Variables

In [38]:
# Assign the string "I love Python" to the variable "my_variable"
my_variable = "I love Python"

In [39]:
my_variable

'I love Python'

In [40]:
# Assign the number 1 to the variable "i"
i = 1

In [41]:
i

1

In [42]:
# Add 1 to "i" and reassign it to "i"
i = i + 1

In [43]:
i

2

# Control of flow 

In [44]:
# Use the "range" funtion to create a list of integers from 0 to 9
range(0,9)

range(0, 9)

In [45]:
list(range(0,9))

[0, 1, 2, 3, 4, 5, 6, 7, 8]

In [46]:
type(list(range(0,9)))

list

In [49]:
type([1,2,3,4])

list

### `for` loops

In [48]:
# create a for loop to print the integers from 0 to 9
for i in range (0,10):
    print(i)

0
1
2
3
4
5
6
7
8
9


In [50]:
for i in range (0,10,2):
    print(i)

0
2
4
6
8


### `if` statement

In [53]:
# Create an if-else statement to print "The value is larger than 5" if the value is larger than 5 and "The value is not larger than 5" if it is not
value = 3
if value > 5:
    print("The value is larger than 5")
else:
    print("The value is not larger than 5") 

The value is not larger than 5


# Quiz Questions

### 1. Simple Calculations

Use Python to calculate solutions to the following expressions.

$$ 3\times8
$$

$$
\frac{(33+7)}{5}\\
$$

In [5]:
# print 3*8
print(3 * 8)

24


In [10]:
# (33+7)/5
print((33 + 7) / 5)



8.0




#### 2. Data Type Comparison


In [12]:
# First expression
result1 = 1 + 2.
print(f"Type of 1 + 2.: {type(result1)}")  # float

# Second expression
result2 = 1 + 2
print(f"Type of 1 + 2: {type(result2)}")   # int

Type of 1 + 2.: <class 'float'>
Type of 1 + 2: <class 'int'>




**Explanation**: 
- `1 + 2.` includes a decimal point, making it a float operation
- `1 + 2` uses integers only, resulting in an integer

#### 3. Square Numbers Loop (n²)


In [13]:
for n in range(1, 6):
    print(f"n = {n}, n² = {n**2}")

n = 1, n² = 1
n = 2, n² = 4
n = 3, n² = 9
n = 4, n² = 16
n = 5, n² = 25




#### 4. Square Numbers Until i² > 1000


In [14]:
# Initialize i
i = 1

# Loop until i² exceeds 1000
while i**2 <= 1000:
    print(f"i = {i}, i² = {i**2}")
    i += 1

# Print final value of i
print(f"\nFinal value of i: {i}")

i = 1, i² = 1
i = 2, i² = 4
i = 3, i² = 9
i = 4, i² = 16
i = 5, i² = 25
i = 6, i² = 36
i = 7, i² = 49
i = 8, i² = 64
i = 9, i² = 81
i = 10, i² = 100
i = 11, i² = 121
i = 12, i² = 144
i = 13, i² = 169
i = 14, i² = 196
i = 15, i² = 225
i = 16, i² = 256
i = 17, i² = 289
i = 18, i² = 324
i = 19, i² = 361
i = 20, i² = 400
i = 21, i² = 441
i = 22, i² = 484
i = 23, i² = 529
i = 24, i² = 576
i = 25, i² = 625
i = 26, i² = 676
i = 27, i² = 729
i = 28, i² = 784
i = 29, i² = 841
i = 30, i² = 900
i = 31, i² = 961

Final value of i: 32




**Explanation**: 
- The loop continues while i² ≤ 1000
- When i = 32, i² = 1024 (exceeds 1000)
- Therefore, the loop stops at i = 32
- 31² = 961 (last value within limit)
- 32² = 1024 (first value exceeding limit)